cmd/server/http/http.go

```diff
@@ -728,3 +726,26 @@ func release(payload *payload, flowSvc *flow.Service) http.HandlerFunc {
 func convertTimeToEpoch(t time.Time) int64 {
 	return t.UnixNano() / int64(time.Millisecond)
 }
+
+type commitInfo struct {
+	AuthorName  string
+	AuthorEmail string
+	Service     string
+}
+
+func extractAuthorFromCommit(message string) (commitInfo, error) {
+	pattern := `^\[(?P<service>.*)\].*\nSigned-off-by:\s(?P<authorName>.*)\s<(?P<authorEmail>.*)>`
+	r, err := regexp.Compile(pattern)
+	if err != nil {
+		return commitInfo{}, errors.WithMessage(err, "regex didn't match")
+	}
+	matches := r.FindStringSubmatch(message)
+	if len(matches) < 1 {
+		return commitInfo{}, errors.New("not enough matches")
+	}
```

Comment:
[`FindStringSubmatch`](https://godoc.org/regexp#Regexp.FindStringSubmatch) returns `nil` in case of no matches so maybe the check should just be that?
As I can see from your tests, it does not look like it will ever return a partial match, so commits missing the `Signed-off-by` line will also result in `nil`.

```suggestion
	matches := r.FindStringSubmatch(message)
	if matches == nil {
		return commitInfo{}, errors.New("no match")
	}
```